src/Laravel/Eloquent/State/LinksHandler.php

```diff
@@ -101,34 +107,44 @@ private function buildQuery(Builder $builder, Link $link, mixed $identifier): Bu
         }
 
         if ($from = $link->getFromProperty()) {
-            $relation = $this->application->make($link->getFromClass());
-            $relationQuery = $relation->{$from}();
-            if (!method_exists($relationQuery, 'getQualifiedForeignKeyName') && method_exists($relationQuery, 'getQualifiedForeignPivotKeyName')) {
+            /** @var Model $relatedInstance */
+            $relatedInstance = $this->application->make($link->getFromClass());
```

Comment:
This line makes me think about https://github.com/api-platform/core/issues/7166, its another subject but do you have any idea on how we could do otherwise to instantiate this class?

Reply:
I think what needs to happen in the item/collection provider is something like `loadMissing` on the relations that are discovered through the eloquent property metadata service. This error happens when calling a relation that hasn't already been hydrated in the class. By calling `loadMissing` we can hopefully prevent that. I'll play around with it and see what I can come up with.